### ISSUE TITLE
Wrong dst loading in city sources

### ISSUE THREAD

**Issue by gondiaz** [next-exp/IC#800]
City sources whose input is a dst-like dataframe have the following structure (see for example `cdst_from_files` or `hits_and_kdst_from_files`)

```
try:
    df = load_dst (path, group, node)
except tb.exceptions.NoSuchNodeError:
    continue
```
where load_dst is the function in `io.dst_io`.  Nevertheless, `load_dst` function outputs a warning in the case of a `NoSuchNodeError` and therefore the intended behaviour of the `try-except` clause in the city source is wrong.

**Comment by mmkekic** [comment]
I'd say that whoever takes this over should also revist all 3 readers since they are all some subset of generic `dst_from_files`. If there is a need to fix the behaviour fast the 2 options are to either catch a warning or upgrade it to error for this scope

**Comment by gondiaz** [comment]
I have proposed a solution in PR #801 which involves removing `NoSuchNodeError` from `load_dst` function. I found it the most natural approach since if a file does not contain the node you are requesting, you get an `Exception` instead of a `warning` which is more error prone.

It might be that your file is wrong and does not contain the node (or you are wrongly requesting a node which is not included in that file type) or you might have an spelling error. Personally, I find more natural to get an `Exception` since I would like to be aware of such a mistake.

Nevertheless, since this is a commonly used function @mmkekic suggests to get input from the likely users: @paolafer @Aretno @jjgomezcadenas @gonzaponte @ausonandres @bpalmeiro @jahernando @msorel @jmalbos @jerenner. 

The `load_dst` function was created by @jjgomezcadenas and @Aretno. The only place in IC that the `warning` is catched is in the function `safe_copy_nexus_eventmap` in `io/mc_io.py`, thus the fact of being a `warning` is not very relevant inside IC.
